.NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/TimexRangeResolver.cs

```diff
@@ -123,12 +123,49 @@ private static IEnumerable<string> ResolveByTimeRangeConstraints(IEnumerable<str
             var resolution = new List<string>();
             foreach (var timex in candidates)
             {
-                var r = ResolveTime(new TimexProperty(timex), collapsedTimeRanges);
-                resolution.AddRange(r);
+                var t = new TimexProperty(timex);
+                if (t.Types.Contains(Constants.TimexTypes.TimeRange))
+                {
+                    var r = ResolveTimeRange(t, collapsedTimeRanges);
+                    resolution.AddRange(r);
+                }
+                else if (t.Types.Contains(Constants.TimexTypes.Time))
+                {
+                    var r = ResolveTime(t, collapsedTimeRanges);
+                    resolution.AddRange(r);
+                }
             }
 
             return RemoveDuplicates(resolution);
         }
+        
+        private static IEnumerable<string> ResolveTimeRange(TimexProperty timex, IEnumerable<TimeRange> constraints)
+        {
+            var candidate = TimexHelpers.TimeRangeFromTimex(timex);
+
+            var result = new List<string>();
+            foreach (var constraint in constraints)
+            {
+                if (TimexConstraintsHelper.IsOverlapping(candidate, constraint))
+                {
+                    var start = Math.Max(candidate.Start.GetTime(), constraint.Start.GetTime());
+                    var time = new Time(start);
+
+                    var resolved = timex.Clone();
```

Comment:
What I really want here is a CloneDateParts because I'm overwriting the time elements and removing the timerange elements. So basically a method to do all that would make some sense. As it is the clone followed by the overwriting has the correct semantics.